File: Medium/Java/1111. Maximum Nesting Depth of Two Valid Parentheses Strings.java
class Solution {
    public int[] maxDepthAfterSplit(String seq) {
        int[] par = new int[seq.length()];
        int count = 0;

        for(int i = 0; i < seq.length(); i++){
            if(seq.charAt(i) == '('){
                count++;
                par[i] = count % 2;
            }else{
                par[i] = count % 2;
                count--;
            }
        }

        return par;
    }
}

File: Medium/Python/1111. Maximum Nesting Depth of Two Valid Parentheses Strings.py
class Solution(object):
    def maxDepthAfterSplit(self, seq):
        par = [0] * len(seq)
        count = 0

        for i in range(len(seq)):
            if seq[i] == '(':
                count += 1
                par[i] = count % 2
            else:
                par[i] = count % 2
                count -= 1
        
        return par
